### PR TITLE
add NetWeaver support for fetching RFC logon password from key vault ...

### PR DESCRIPTION
**Pull request by MikeFrei**
…secret URL.  Also fix issue exposed by recent sapmon.py execution loop changes that will result in unhandle exception for datetime comparision between timezone aware and naive values for lastRunLocal.  Fix is to make base.py providerCheck.isDue() support both naive and aware comparison.

### CODE REVIEW

File: sapmon/payload/provider/sapnetweaver.py [hunk 1]
```diff
@@ -109,12 +109,41 @@ def parseProperties(self) -> bool:
...
         self.sapUsername = self.providerProperties.get('sapUsername', None)
         self.sapPassword = self.providerProperties.get('sapPassword', None)
         self.sapClientId = self.providerProperties.get('sapClientId', None)
         self.sapRfcSdkBlobUrl = self.providerProperties.get('sapRfcSdkBlobUrl', None)
 
+        # if user did not specify password directly via UI, check to see if they instead
+        # provided link to Key Vault secret
+        if not self.sapPassword:
+            sapPasswordKeyVaultUrl = self.providerProperties.get("sapPasswordKeyVaultUrl", None)
+            if sapPasswordKeyVaultUrl:
```
Comment (PakDLiu) [inline sapmon/payload/provider/sapnetweaver.py:121]: Should there be a case where if neither are set?

**Review by PakDLiu — APPROVED** [review]

File: sapmon/payload/provider/sapnetweaver.py (hunk 1 above)
Comment (MikeFrei) [reply sapmon/payload/provider/sapnetweaver.py:121]: that is already handled, if neither are set then self.sapPassword will end up with default value of None, and RFC metric actions will be disabled

Comment (MikeFrei) [reply sapmon/payload/provider/sapnetweaver.py:121]: we are still treating RFC metric collection as optional

Comment (PakDLiu) [reply sapmon/payload/provider/sapnetweaver.py:121]: I see, minior issue: should we have a log statement stating neither was set?
I guess the frontend will prevent that

**Pull request merged**